conan/tools/cmake/cmake.py

```diff
@@ -83,8 +83,13 @@ def configure(self, variables=None, build_script_folder=None):
             pkg_folder = self._conanfile.package_folder.replace("\\", "/")
             arg_list.append('-DCMAKE_INSTALL_PREFIX="{}"'.format(pkg_folder))
         if platform.system() == "Windows" and self._generator == "MinGW Makefiles":
-            # It seems this doesn't work in the toolchain file, it needs to be here in command line
+            # It seems these don't work in the toolchain file, they need to be here in command line
             arg_list.append('-DCMAKE_SH="CMAKE_SH-NOTFOUND"')
+            cmake_make_program = self._conanfile.conf.get("tools.gnu:make_program", default=None)
```

Comment:
I have had a second look, just in case, and it really seems difficult.
So I am approving and merging this. However, I think it would be better if we started storing this arguments in ``CMakeToolchain`` in the ``conanbuild.conf`` file, or even better, if we start considering moving to a CMakePresets.json compatible format.